### PR TITLE
Fix event.location token to include supplemental address details, use in event receipts

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix event.location token to include supplemental address details, use in event receipts

Before
----------------------------------------
`$location` is notice prone in event receipts & requires a lot of work on the form layer

After
----------------------------------------
`{event.location}` is used. We didn't add this earlier cos I needed to check if losing the vcards mattered but I questioned that via the dev-digest & there was no case made to keep it (& some to remove) - https://lab.civicrm.org/dev/core/-/issues/4366

The only other gap was the missing supplemental field data- which this adds

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I just need to update CRM_Utils_TokenConsistencyTest::testEventTokenConsistency

**Comment by larssandergreen** [comment]
If you enable all the address field in Address Settings, you also have county, postal code suffix, lat/long and country in manage event — and I think if the field appears there, the user has a reasonable expectation that whatever they enter is going to come out the other end. So I think we need to support the full list (except I'd say we don't need country because you have to select country to get province and you probably don't want actually to display it). Maybe lat/long is a little silly, but if we show the field and then don't output it, that seems like a bug.

**Comment by eileenmcnaughton** [comment]
OK - so I think I should load all fields, convert the custom ones to 'custom_' format & pass them to the function.. & it can filter

**Comment by eileenmcnaughton** [comment]
@larssandergreen Ok- I've broadened the support. I was gonna actually tackle custom address fields too but some of those rabbits are pretty bitey so I figured I could beat a hasty retreat & recoup before tackling that. It's not needed for parity

File: CRM/Event/Tokens.php [hunk 1]
```diff
@@ -205,12 +203,16 @@ protected function getEventTokenValues(int $eventID = NULL): array {
           'custom.*',
         ], $this->getExposedFields()))
         ->execute()->first();
-      $tokens['location']['text/plain'] = \CRM_Utils_Address::format([
-        'street_address' => $event['loc_block_id.address_id.street_address'],
-        'city' => $event['loc_block_id.address_id.city'],
+      $addressValues = [
+        'address_name' => $event['loc_block_id.address_id.name'],
         'state_province' => $event['loc_block_id.address_id.state_province_id:label'],
```
Comment (larssandergreen) [inline CRM/Event/Tokens.php:208]: County doesn't work, assume we need a similar line to get that label.

**Comment by larssandergreen** [comment]
Other than the county, this looks good to me and works.

File: CRM/Event/Tokens.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Event/Tokens.php:210]: @larssandergreen I just added country & county (although I haven't tested them yet)

Comment (larssandergreen) [reply CRM/Event/Tokens.php:210]: I think better to leave country out. I can't imagine a context where you actually want to print the country in the event receipt (presumably registrants know which country the event is in) and you can't select a province without selecting a country, so users can't stop it from displaying if they need a province.

**Comment by eileenmcnaughton** [comment]
Added merge-on-pass as @larssandergreen had OK-d this pending fix to the county & I pushed up test fix too

**Pull request merged**